api/client.go

```diff
@@ -309,6 +311,9 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 	if err := c.validateCharmVersion(ch); err != nil {
 		return nil, errors.Trace(err)
 	}
+	if err := lxdprofile.ValidateCharmLXDProfile(ch); err != nil {
```

Comment:
perhaps the tests (TestLXDProfileLocalCharm, TestLXDProfileLocalCharmFails) in cmd/juju/application/deploy_test.go should move to client_test.go?